src/amp-url-creator.js

```diff
@@ -58,25 +58,32 @@ const MAX_DOMAIN_LABEL_LENGTH_ = 63;
  * @param {string} url The complete publisher url.
  * @param {object} initParams Params containing origin, etc.
  * @param {string} opt_cacheUrlAuthority
- * @param {string} opt_viewer_js_version
- * @return {string} the Cache Url.
+ * @param {string} opt_viewerJsVersion
+ * @return {!Promise<string>}
  * @private
  */
 export function constructViewerCacheUrl(url, initParams,
-  opt_cacheUrlAuthority, opt_viewer_js_version) {
+  opt_cacheUrlAuthority, opt_viewerJsVersion) {
   const parsedUrl = parseUrl(url);
-  const cacheDomain = constructCacheDomainUrl_(url, opt_cacheUrlAuthority);
   const protocolStr = parsedUrl.protocol == 'https:' ? 's/' : '';
-  const viewerJsVersion = opt_viewer_js_version ? opt_viewer_js_version :
+  const viewerJsVersion = opt_viewerJsVersion ? opt_viewerJsVersion :
     DEFAULT_VIEWER_JS_VERSION_;
 
-  return cacheDomain + 
+  return new Promise(resolve => {
+    constructCacheDomainUrl_(parsedUrl.host, opt_cacheUrlAuthority).then(cacheDomain => {
+      resolve(
+        parsedUrl.protocol + '//' +
+        cacheDomain + 
           '/v/' +
           protocolStr +
           parsedUrl.host + 
-          '/?amp_js_v=' + viewerJsVersion +
+          parsedUrl.pathname +
+          '?amp_js_v=' + viewerJsVersion +
```

Comment:
I think this will drop query parameters in the AMP URL:
http://www.example.com/foo?amp=true
Should produce:
https://cdn.ampproject.org/v/www.example.com/foo?amp=true&amp_js_v=0.1